experimental/sketching.py

```diff
@@ -0,0 +1,166 @@
+from jax import random
+from jax import numpy as np
+from neural_tangents._src.utils import dataclasses
+from typing import Optional, Callable
+
+
+# TensorSRHT of degree 2. This version allows different input vectors.
+@dataclasses.dataclass
+class TensorSRHT2:
+
+  input_dim1: int
+  input_dim2: int
+  sketch_dim: int
+
+  rng: np.ndarray
```

Comment:
Here and elsewhere, `jax.random.KeyArray` is a slightly more explicit type for RNGs.